include/dt-bindings/gpio/gpio.h

```diff
@@ -1,180 +1,429 @@
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_EDGE   (1 << (GPIO_INT_POS + 1))
 
+/** Trigger detection when input state is (or transitions to) low.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_LOW    (1 << (GPIO_INT_POS + 2))
+
+/** Trigger detection on input state is (or transitions to) high.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_HIGH   (1 << (GPIO_INT_POS + 3))
+
+/** Interrupt triggers on both rising and falling edge.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ *
+ * For compatibility with old code this should be combined with
+ * `GPIO_INT_EDGE`.
+ *
+ * For compatibility with new code this should be combined with both
+ * `GPIO_INT_LOW` and `GPIO_INT_HIGH`.  Updated driver code should
+ * prefer to check the individual flags.
+ *
+ * @deprecated Replace tests with `GPIO_INT_LOW | GPIO_INT_HIGH`.
+ *
+ * @{
+ */
+#define GPIO_INT_DOUBLE_EDGE    (1 << (GPIO_INT_POS + 4))
+
+/** GPIO pin trigger on rising edge. */
+#define GPIO_INT_EDGE_RISING    (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_HIGH)
+
+/** GPIO pin trigger on rising edge. */
+#define GPIO_INT_EDGE_FALLING   (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_LOW)
+
+/** GPIO pin trigger on rising or falling edge. */
+#define GPIO_INT_EDGE_BOTH      (GPIO_INT_ENABLE \
+				 | GPIO_INT_EDGE \
+				 | GPIO_INT_LOW	 \
+				 | GPIO_INT_HIGH \
+				 | GPIO_INT_DOUBLE_EDGE)
+
+/** GPIO pin trigger on level low. */
+#define GPIO_INT_LEVEL_LOW      (GPIO_INT_ENABLE  \
+				 | GPIO_INT_LEVEL \
+				 | GPIO_INT_LOW)
+
+/** GPIO pin trigger on level high. */
+#define GPIO_INT_LEVEL_HIGH     (GPIO_INT_ENABLE  \
+				 | GPIO_INT_LEVEL \
+				 | GPIO_INT_HIGH)
+
+/** @} */
 
 /**
- * @name GPIO pull flags
- * The `GPIO_PUD_*` flags are used with `gpio_pin_configure` or `gpio_port_configure`,
- * to specify the pull-up or pull-down electrical configuration of a GPIO pin.
+ * @name GPIO polarity flags
+ *
+ * The `GPIO_POL_*` flags are used with `gpio_pin_configure` or
+ * `gpio_port_configure`, to specify the polarity of a GPIO pin.
+ *
  * @{
  */
 /** @cond INTERNAL_HIDDEN */
-#define GPIO_PUD_POS		8
+#define GPIO_POL_POS            14
+#define GPIO_POL_MASK           (1 << GPIO_POL_POS)
 /** @endcond */
 
-/** Pin is neither pull-up nor pull-down. */
-#define GPIO_PUD_NORMAL		(0 << GPIO_PUD_POS)
+/** GPIO pin polarity is normal. */
+#define GPIO_POL_NORMAL         (0 << GPIO_POL_POS)
 
-/** Enable GPIO pin pull-up. */
-#define GPIO_PUD_PULL_UP	(1 << GPIO_PUD_POS)
+/** GPIO pin polarity is inverted. */
```

Comment:
Maybe it would be good to clarify on the occasion what the polarity inversion means? I recall I found it a bit confusing when reviewing the updated nRF GPIO driver, and I'm still not sure if I got it right. I assume that it's that bit of value 1 indicates the low state and bit of value 0 the high one, and this concerns reading inputs and writing outputs. But what about the interrupt triggers, are they also supposed to be applied inversely?

Reply:
I'm just preserving the original comment (though that's not clear from the way the diff gets extracted).  I don't believe the impact of using the flag can or should be generically described.  The SX1509B has this feature, and it may not have the same behavior as Nordic or other devices (as I found with the drive strength flags, where the described behavior was wrong for SX1509B).

I will update the description with a note that the impact of the flag is driver-specific, including that it may be ignored.